### ISSUE TITLE
Implement "Riders" Narrative

### ISSUE THREAD

**Issue by hobuobi** [Recidiviz/public-dashboard#518]
All incarcerated individuals in prison in Idaho can be divided into two groups: riders (those who were sentenced to a special short-term in prison by a judge) and termers (those who were sentenced normally to a period longer than one year). 

To highlight the program and its successes and failures, we will implement the "Rider Program" narrative on Spotlight ID. This will be a special page separate from the core narratives (similar to the Racial Disparities narrative). Designs for this can be found in this [Figma file.](https://www.figma.com/file/up87UsbvSmAPAIxgdPxFXW/Spotlight-Idaho?node-id=0%3A1).

The new page should include the following. The page should also be accessible from the "Data Narratives" dropdown in the navigation. 
- **[P0] [text-only] An introduction with a brief explanation of the Rider program**
    - Final copy forthcoming; example copy in the Figma file. 
- **[P0] How prevalent are “riders” in the IDOC population?**  
    - Metric: Population Over Time 
    - Viz: Area Chart, split by "Riders" and "Regular Incarceration Term" (similar to [Prison Population Over Time](https://dashboard.docr.nd.gov/us-nd/narratives/prison/3))
    - Breakdowns: N/A
- **[P0] Who is on a Rider in Idaho?** 
    - Metric: Current population of Rider, Current population of (Regular Incarceration) Termers
    - Visualization: Double bar chart (similar to ["How does sentencing impact disparities"](https://dashboard.docr.nd.gov/us-nd/narratives/racial-disparities/3))
    - Breakdown: Race, Age, Gender 
- **[P1] Why are people admitted for Rider terms?** 
    - Metric: Original Charge (for those admitted in the last year)
    - Visualization: Bubble Chart / Bar Charts (similar to ["How did they get there?"](https://dashboard.docr.nd.gov/us-nd/narratives/prison/4)) 
- **[P0] What do outcomes look like for riders?** 
    - Metric: Reincarceration Rate (1 year cohort with three year outcomes) 
    - Viz: 
        - OPTION 1 (implement first): Line Chart (similar to ["How many people end up back in prison?"](https://dashboard.docr.nd.gov/us-nd/narratives/prison/7))
        - OPTION 2 (P1): Grouped Bars (have never implemented this before on Spotlight, but would show 3 vertical bars per category in a breakdown). 
    - Breakdowns: Race, Age, Gender
- **[P0] [text-only] Policy Recommendation:** Implement more normalized treatment outside of prison 
    - Copy forthcoming.

A separate ticket will be created for the data analysis and views that power this narrative. In the meantime, the mock data in this file can provide a starting point for data visualizations; if these don't work or are missing for whatever reason, contact Humphrey and use data downloaded from similar charts in dashboard.docr.nd.gov to unblock. 

Contact Humphrey with any questions.

**Comment by macfarlandian** [comment]
After some study and discussion with @hobuobi I would offer the following suggestions for how to implement the sections described here: 

### How prevalent are riders (population over time)
This should be a new Metric type as well as a new Record type, "HistoricalPopulationByCategory" or something like that. Instead of the pre-defined demographic breakdown fields, it should use a single `category` field to split the records into multiple data series. At the moment it only needs to support the time window filter, not any of the other standard ones.

You should be able to use the same chart component for this but you will also need a new `VizHistoricalPopulationByCategory` component that is included in the `VizMapper`.

### Who is on a rider? (dual bars)
This should also be a new Metric type: `CategoriesByDemographicMetric`, which is basically the inverse of `DemographicsByCategoryMetric` and uses the same underlying `DemographicsByCategoryRecord` type. Instead of grouping the records by the selected demographic field, it would group them by the `category` and then filter them by the demographic category. There would be no "ALL" option here, you would have to set a default demographic view (I would choose race/ethnicity).

The required visualization is functionally identical to what you see in [`VizDemographicsByCategory`](https://github.com/Recidiviz/public-dashboard/blob/nasaownsky/518-riders-narrative/spotlight-client/src/VizDemographicsByCategory/VizDemographicsByCategory.tsx#L88) when a demographic view is selected; I think you could factor out this block of code that renders one proportional bar per data series and then reuse it in a new `VizCategoriesByDemographic` component.

I am not sure that the suggested alternative (reusing the multi-chart display that shows all demographic breakdowns at once) is really worth the effort as an interim step, since that metric/component only knows how to filter data by location. You would have to do some work to generalize it anyway and if it doesn't even show the data in the way we want then I personally would not bother, it will just make it take longer to actually do it the right way.

### Why are people admitted (bubble chart)
As Humphrey says, the existing Metric and viz component should work fine for this, it will just be a new data source and data transformation function (all of this work is done in the `createMetricMapping` module.

### What do outcomes look like (reincarceration)
This should be yet another new data type, `RateByCategoryAndDemographics` or something like that. It can be pretty generic, the record format is basically just the union of `DemographicFields` and `RateFields` plus a `category` field. The metric and viz component could be similarly generic. You will, in fact, need an entirely new chart component for this one also, we don't currently use a grouped bar chart like this anywhere else in Spotlight ([Semiotic supports it just fine, though](https://semiotic.nteract.io/guides/bar-chart#cluster--grouped-bar-chart-using-the-same-data-model)).  

I actually would not recommend trying to use the existing Recidivism metric/components for this, even as an interim step. It is both visually inappropriate (since we don't actually have data over time) and not well supported by the implementation (it only supports cohorts that are defined by years, not any arbitrary value).
